transports/tcp/src/provider/async_io.rs

```diff
@@ -39,7 +39,8 @@ use std::task::{Context, Poll};
 /// # #[async_std::main]
 /// # async fn main() {
 /// let mut transport = tcp::async_io::Transport::new(tcp::Config::default());
-/// let id = transport.listen_on("/ip4/127.0.0.1/tcp/0".parse().unwrap()).unwrap();
+/// let id = Default::default();
```

Comment:
I'd prefer to not use `Default` here.

Reply:
I do plan on going back to change the `Default` very soon as I dont prefer using them but wanted to make sure the test were resolved.